### DISCUSSION TITLE
less ram usage

### DISCUSSION THREAD

**Discussion by Josiah-tan** [opensteno/plover#1570] [Technical Discussion]
In regards to trying to get plover on something like an [ox64](https://liliputing.com/pine64-ox64-is-a-raspberry-pi-pico-sized-single-board-pc-with-a-risc-v-processor/) (less than 64MB), is it possible to reduce the amount of RAM required to run plover?

**Comment by user202729** [comment]
A bit difficult.

Last time I try the bottlenecks are mostly the American word list (removing this won't affect the result too much) and the dictionary itself (main.json is merely 4MB, but Python small string overhead means it gets much larger? Maybe try your luck with C extension or something).

----

(Actually this is probably solvable in pure Python by packing everything into a bytearray then binary search on it for dictionary lookup...? But that sounds more hacky than just implementing a dictionary in C)

**Comment by Josiah-tan** [comment]
![image](https://user-images.githubusercontent.com/65082231/201474230-65917b42-93aa-499b-97ef-26d9b962e5c2.png)
Strangely, even when testing without dictionaries, Plover still uses a relatively large amount of ram

**Reply by user202729** [reply]
Yes as I mentioned earlier, try deleting `assets/american_english_words.txt`? (may need running from source)

**Reply by Josiah-tan** [reply]
![image](https://user-images.githubusercontent.com/65082231/201474968-c9c03d07-d556-4513-9046-76a9ef424ddd.png)
ran from source on linux but I'm not sure if there's much difference

**Comment by benoit-pierre** [comment]
Run without the GUI, use SQLite dictionaries, don't load the wordlist…

| VIRT |   RES |   SHR | |
|------|-------|-------|-------
| 858M |  206M | 55920 | normal
| 420M |  140M |  8564 | no GUI (PyQt5 not installed)
| 446M | 83688 |  9636 | no GUI, SQLite dictionaries
| 323M | 28360 |  9504 | no GUI, SQLite dictionaries, no wordlist

**Reply by user202729** [reply]
Ah right, I wrote that one before. Even I forgot.

https://github.com/user202729/plover-sqlite-dictionary

These days my computer has sufficient RAM that I don't need the plugin.

**Reply by Josiah-tan** [reply]
Is there a way to get a wheel for plover without having PyQt5 installed and without the wordlist?
Running plover from source on a raspberry pi is very difficult (not sure if the script that Benoit gave works because plover now uses tox)
